plugins/genesys/fu-genesys-plugin.c

```diff
@@ -35,13 +36,52 @@ fu_genesys_plugin_constructed(GObject *obj)
 	fu_context_add_quirk_key(ctx, "GenesysUsbhubSwitchRequest");
 	fu_context_add_quirk_key(ctx, "GenesysUsbhubWriteRequest");
 	fu_plugin_add_device_gtype(plugin, FU_TYPE_GENESYS_USBHUB_DEVICE);
+	fu_plugin_add_device_gtype(plugin, FU_TYPE_GENESYS_HUBHID_DEVICE);
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_GENESYS_USBHUB_FIRMWARE);
 	fu_plugin_add_firmware_gtype(plugin, NULL, FU_TYPE_GENESYS_SCALER_FIRMWARE);
 }
 
+static void
+fu_genesys_plugin_device_added(FuPlugin *self, FuDevice *device)
+{
+	GUsbDevice *gusb_parent = NULL;
+
+	/* link hid to parent hub */
+	if (!FU_IS_GENESYS_HUBHID_DEVICE(device))
+		return;
+
+	gusb_parent = g_usb_device_get_parent(fu_usb_device_get_dev(FU_USB_DEVICE(device)));
+	if (gusb_parent != NULL) {
+		GPtrArray *devices = fu_plugin_get_devices(self);
+		FuDevice *parent = NULL;
+
+		for (guint i = 0; i < devices->len; i++) {
+			FuDevice *dev = g_ptr_array_index(devices, i);
+
+			if (!FU_IS_GENESYS_USBHUB_DEVICE(dev))
+				continue;
+
+			if (fu_usb_device_get_dev(FU_USB_DEVICE(dev)) == gusb_parent) {
+				parent = dev;
+				break;
+			}
+		}
```

Comment:
i think split this out to something like `FuDevice *fu_genesys_plugin_get_device_by_gusb_device(self, GUsbDevice *gusb_device)`